obal/data/playbooks/library/rpmspec_changelog_check.py

```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/python
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+from ansible.module_utils.basic import AnsibleModule
+import os
+import subprocess
+import fnmatch
+
+def run_module():
+    module_args = dict(
+        directory=dict(type='str', required=True),
+    )
+
+    result = dict(
+        changed=False,
+        changelog_version_release='',
+        specfile_version_release=''
+    )
+
+    module = AnsibleModule(
+        argument_spec=module_args,
+        supports_check_mode=True
+    )
+
+    if module.check_mode:
+        return result
+
+    specdir = module.params['directory']
+
+    specfile = [ os.path.join(specdir, filename) for filename in os.listdir(specdir) if fnmatch.fnmatch(filename, "*.spec") ][0]
+    changelog_top = subprocess.check_output(['rpm', '--query', '--changelog', '--specfile', specfile]).split("\n")[0]
+    changelog_version_release = changelog_top.split(" ")[-1]
+
+    specfile_version_release = subprocess.check_output(['rpm', '--query', '--queryformat', '%{version}-%{release}\n', '--specfile', specfile]).split("\n")[0]
+    specfile_version_release = specfile_version_release.split(".")[:-1] # trim off the dist
```

Comment:
In other places I explicitly set the dist to empty. I think I used `rpmspec --undefine dist` which removes the need for splitting and joining.

Reply:
oh my goodness that's so much easier!